src/hackney_pool.erl

```diff
@@ -521,25 +508,6 @@ add_to_queue(Connection, From, Ref, Requester, Queues) ->
       dict:store(Connection, queue:in({From, Ref, Requester}, Q), Queues)
   end.
 
-%------------------------------------------------------------------------------
-%% @private
-%%------------------------------------------------------------------------------
-del_from_queue(Connection, Ref, Queues) ->
```

Comment:
This is not needed anymore. Moreover, there is `remove_pending` which can be used instead and is implemnted in a more efficient manner: first checking `pending` dict and (if found) filter `queue`.